pkg/cache/clusterqueue.go

```diff
@@ -407,10 +403,14 @@ func (c *clusterQueue) updateWithAdmissionChecks(checks map[kueue.AdmissionCheck
 	}
 }
 
-func (c *clusterQueue) addWorkload(w *kueue.Workload) error {
+func (c *clusterQueue) addWorkload(w *kueue.Workload, allowUpdate bool) error {
```

Comment:
Can we rename/add comments to this function so it better expresses intentions of updating a Workload? Or maybe we could even add a separate `updateWorkload` function. WDYT?

Reply:
For sure, I will give it a pass to update the function names and comments.

Reply:
In the most recent update I propose to rename the function as addOrUpdateWorkload. This name is more consistent with other places in code I think.